### DISCUSSION TITLE
Unable to find Flutter version 3.10.7

### DISCUSSION THREAD

**Discussion by abelokon0711** [codemagic-ci-cd/codemagic-docs#1991] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Set the Flutter version in the environment to 3.10.7.

Either directly:

```
    environment:
      flutter: 3.10.7
```

Or by using `fvm`:

```
    environment:
      flutter: fvm
```

Given the following `fvm_config.json`:

```
{
  "flutterSdkVersion": "3.10.7",
  "flavors": {}
}
```

### Expected results

Build uses the Flutter hotfix version 3.10.7.

### Actual results

We get the following error:

```
== Use Flutter version 3.10.7 ==

Build failed :|
Unable to find Flutter version 3.10.7
```

**Comment by helinanever** [comment]
Hi @abelokon0711, the Flutter version you are trying to use does not exist, hence Codemagic is unable to find it. See the available Flutter versions here: https://docs.flutter.dev/release/archive?tab=macos

**Reply by blaugold** [reply]
It does exist. 🙂 Here is the announcement: https://groups.google.com/g/flutter-announce/c/lDixv4uZ1zs

**Reply by blasko** [reply]
https://github.com/flutter/flutter/wiki/Upgrading-from-3.10.6-to-3.10.7

Issue found at: https://github.com/flutter/flutter/issues/131319

Announcement: https://groups.google.com/g/flutter-announce/c/lDixv4uZ1zs/m/oT5vD0-UAgAJ?utm_medium=email&utm_source=footer
[2:13](https://nevercode.slack.com/archives/C05DJ6M6797/p1692724423736199)

Seems like a release management issue unfortunately, from the Google/Flutter side at the moment.

**Reply by blaugold** [reply]
> Seems like a release management issue unfortunately, from the Google/Flutter side at the moment.

In what sense? `fvm install 3.10.7`, for example, works fine.

**Comment by remarkov** [comment]
@blasko is right, it is a flutter release management issue. The builder runner is getting the list of available versions [here](https://storage.googleapis.com/flutter_infra_release/releases/releases_macos.json) and 3.10.7 is not listed there, that's why it fails.
If you need 3.10.7 on codemagic, you may specify 3.10.6 in your config and add a custom script as per Kevin's upgrade instruction [here](https://github.com/flutter/flutter/wiki/Upgrading-from-3.10.6-to-3.10.7).

UPD: here is a workflow sample for your reference:

```  
workflow-name:
  ...
  environment:
    flutter: 3.10.6
  scripts:
    - name: Upgrade flutter to 3.10.7
      script: |
        cd ~/programs/flutter
        git fetch --tags
        git checkout 3.10.7
        flutter --version
  ...
```

**Reply by blaugold** [reply]
Thanks for the workflow sample! I was just wondering where Flutter is installed on the build runners :)

**Comment by helinanever** [comment, marked as answer]
**UPDATE:** Flutter `3.10.7` is now available on Codemagic simply by selecting the version in the Flutter workflow editor or by specifying `flutter: 3.10.7` in codemagic.yaml. The script to upgrade Flutter is no longer required (but if you have already added it, it will still work).

**Comment by RicoLiu** [comment]
> **UPDATE:** Flutter `3.10.7` is now available on Codemagic simply by selecting the version in the Flutter workflow editor or by specifying `flutter: 3.10.7` in codemagic.yaml. The script to upgrade Flutter is no longer required (but if you have already added it, it will still work).

@helinanever Why Flutter Stable channel do not display `flutter: 3.10.7` version?
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/12722992/942c401c-7c18-444b-96d8-d3b15c751b8a)

**Reply by helinanever** [reply]
@RicoLiu That's because `3.10.7` was never released to the stable channel, they released this version as a tag.
